src/bindings/c/src/ov_remote_context.cpp

```diff
@@ -0,0 +1,208 @@
+                                   ov_remote_context** context,
+                                   ...) {
+    if (!core || !device_name || !context || property_args_size % 2 == 1) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+
+    try {
+        ov::AnyMap property = {};
+        size_t property_size = property_args_size / 2;
+        va_list args_ptr;
+        va_start(args_ptr, context);
+        GET_INTEL_GPU_PROPERTY_FROM_ARGS_LIST(property_size);
+        va_end(args_ptr);
+
+        std::string dev_name = device_name;
+        ov::RemoteContext object = core->object->create_context(dev_name, property);
+
+        std::unique_ptr<ov_remote_context> _context(new ov_remote_context);
+        _context->object = std::make_shared<ov::RemoteContext>(std::move(object));
+        *context = _context.release();
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_remote_context_create_tensor(const ov_remote_context_t* context,
+                                            const ov_element_type_e type,
+                                            const ov_shape_t shape,
+                                            const size_t property_args_size,
+                                            ov_tensor_t** remote_tensor,
+                                            ...) {
+    if (!context || !shape.dims || !remote_tensor || property_args_size % 2 == 1) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+
+    try {
+        ov::AnyMap property = {};
+        size_t property_size = property_args_size / 2;
+        va_list args_ptr;
+        va_start(args_ptr, remote_tensor);
+        GET_INTEL_GPU_PROPERTY_FROM_ARGS_LIST(property_size);
+        va_end(args_ptr);
+
+        ov::Shape tmp_shape;
+        std::copy_n(shape.dims, shape.rank, std::back_inserter(tmp_shape));
+        auto tmp_type = get_element_type(type);
+        ov::RemoteTensor object = context->object->create_tensor(tmp_type, tmp_shape, property);
+
+        std::unique_ptr<ov_tensor> _remote_tensor(new ov_tensor);
+        _remote_tensor->object = std::make_shared<ov::RemoteTensor>(std::move(object));
+        *remote_tensor = _remote_tensor.release();
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_remote_context_get_device_name(const ov_remote_context_t* context, char** device_name) {
+    if (!context || !device_name) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+
+    try {
+        auto value = context->object->get_device_name();
+        *device_name = str_to_char_array(value);
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+//!< Output ov::AnyMap to be C string style data:
+//!<    "key_name_1:key_value_str_1;key_name_2:key_value_str_2;..."
+inline void convert_params_to_string(ov::AnyMap& paramsMap, char*& res_str, size_t& size) {
+    std::string res = "";
+    for (auto& param : paramsMap) {
+        res += param.first;
+        res += ":";
```

Comment:
It is hard to say whether is aligned with C++, because C++ will return ov::AnyMap which contains key and value for remoteContext param, but C interface will return a single string which put key and value together, so we should have a format to separate key and value.

For example:  
      C++ return ov::AnyMap = {{"CONTEXT_TYPE" , "OCL"}, {"OCL_CONTEXT", "0x5588d21a18a0"} , {"OCL_QUEUE", "0"}}
      
      and C return:
             "CONTEXT_TYPE:OCL;OCL_CONTEXT:0x5588d21b2aa0;OCL_QUEUE:0;"
             it uses ":" to separate key and value, and use ";" to separate different property

Reply:
Can you try:
```
ov::Any any = any_map;
std::cout << "ov::AnyMap = " << any.as<std::string>() << std::endl;
```

Reply:
This method does print string as below:
`ov::AnyMap = CONTEXT_TYPE OCL OCL_CONTEXT 0x5592bf31d570 OCL_QUEUE 0x5592bf016bd0`

There is space separator between each value, do we change to be this format?